Library/Homebrew/cmd/developer.rb

```diff
@@ -0,0 +1,53 @@
+# typed: true
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  extend T::Sig
+
+  module_function
+
+  sig { returns(CLI::Parser) }
+  def developer_args
+    Homebrew::CLI::Parser.new do
+      description <<~EOS
+        Control Homebrew's developer mode. When developer mode is enabled,
+        `brew update` will update Homebrew to the latest commit on the `master`
+        branch instead of the latest stable version.
+
+        `brew developer` [`state`]:
+        Display the current state of Homebrew's developer mode.
+
+        `brew developer` (`on`|`off`):
+        Turn Homebrew's developer mode on or off respectively.
+      EOS
+
+      named_args %w[state on off], max: 1
+    end
+  end
+
+  def developer
+    args = developer_args.parse
+
+    case args.named.first
+    when nil, "state"
+      if Homebrew::EnvConfig.developer?
+        puts "Developer mode is enabled because #{Tty.bold}HOMEBREW_DEVELOPER#{Tty.reset} it set."
+      elsif Homebrew::Settings.read("devcmdrun") == "true"
```

Comment:
I wonder if we want a dedicate method for this that takes into account the above? Also, need to handle the case where `HOMEBREW_UPDATE_TO_TAG` is set.

Reply:
Can you explain what you mean by "dedicate method" in this case?

Also, what does `HOMEBREW_UPDATE_TO_TAG` do? Is it to turn off all of the developer stuff or just the updating? I figure when running `brew developer off` I can just output a message if it's set similar to what I've done with `HOMEBREW_DEVELOPER`

Reply:
Sorry I mean a dedicated method that is something like `Homebrew::Settings.devcmdrun?` or something on another class.
> Also, what does `HOMEBREW_UPDATE_TO_TAG` do? Is it to turn off all of the developer stuff or just the updating?

Just updating.



> I can just output a message if it's set similar to what I've done with `HOMEBREW_DEVELOPER`

Sounds good 👍🏻

Reply:
> Sorry I mean a dedicated method that is something like `Homebrew::Settings.devcmdrun?` or something on another class.
> 
> >

At the moment, this is the only time we ever use ruby to read `devcmdrun`, so I'm not sure it's necessary.

Reply:
Gotcha, fine with me then 👍🏻